ext/mysqli/mysqli_api.c

```diff
@@ -647,28 +647,118 @@ PHP_FUNCTION(mysqli_stmt_fetch)
 /* }}} */
 
 /* {{{  php_add_field_properties */
-static void php_add_field_properties(zval *value, const MYSQL_FIELD *field)
-{
-	add_property_str(value, "name", zend_string_copy(field->sname));
-
-	add_property_stringl(value, "orgname", (field->org_name ? field->org_name : ""), field->org_name_length);
-	add_property_stringl(value, "table", (field->table ? field->table : ""), field->table_length);
-	add_property_stringl(value, "orgtable", (field->org_table ? field->org_table : ""), field->org_table_length);
-	add_property_stringl(value, "def", (field->def ? field->def : ""), field->def_length);
-	add_property_stringl(value, "db", (field->db ? field->db : ""), field->db_length);
+static void php_add_field_properties(zval *z_object, const MYSQL_FIELD *field)
```

Comment:
This does not look like an improvement :man_shrugging: What's bad about the old API for these cases (except for the name)?

Reply:
The main issue is that the ``_str()`` variant takes ownership of the zend_string* value.

The other motivation is to be able to possibly use known zend_strings for the property name access.